Lighting/Inc/lighting_controller.hpp

```diff
@@ -131,22 +192,58 @@ class LightingController {
 	 * @param colour : desired colour
 	 * @param brightness : desired brightness
 	 */
-	void set_domain_colour(ControlDomain domain, RGB_colour_t colour,
+	void set_domain_colour_and_brightness(ControlDomain domain, RGB_colour_t colour,
 			uint8_t brightness);
 
+	/**
+	 * This will set just the brightness of all the LEDs that are part of a control domain.
+	 *
+	 * @param domain : domain to be modified
+	 * @param colour : desired colour
+	*/
+	void set_domain_colour(ControlDomain domain, RGB_colour_t colour);
+
+	/**
+	 * This will set just the colour of all the LEDs that are part of a control domain.
+	 *
+	 * @param domain : domain to be modified
+	 * @param brightness : desired brightness
+	*/
+	void set_domain_brightness(ControlDomain domain, uint8_t brightness);
+
+	/**
+	 * This is a state machine that changes enabled leds, control domain colour
+	 * and brightness based on the state the drone is in.
+	 *
+	 * @param input : input that determines the next drone state and output.
+	 */
+	void transition_to(uint8_t input);
+
+
+	/*
+	 * Returns the state of the drone
+	 */
+	State get_drone_state();
+
 	// TODO: add function to "recolour domain" (enables & sets colour)
 	// Use this instead of set + enable?
 
 private:
-	static constexpr uint8_t NUM_LEDS = 6;
+	static constexpr uint8_t NUM_LEDS = 10;
+	State drone_state; //state of the drone.
+
 	uint8_t *dma_buffer;
 	uint8_t *bank_buffer;
 	WS2812 *leds;
-	bool domain_state[CD_LENGTH];				// true when domain is active
-	uint8_t domain_leds[CD_LENGTH];	// Bitmask of LED's which are active in each domain
-	RGB_colour_t domain_colours[CD_LENGTH];		// Domain colour
-	uint8_t domain_brightness[CD_LENGTH];		// Domain brightness
 
+	uint8_t domain_allowed;					// true when a domain is allowed to be active
+	uint8_t domain_active;					// true when domain is active
```

Comment:
It would be nice if  we could get an explicit comment that these is a **bitfield** of allowed domains, otherwise it could be a bit confusing to read.